umm-spec-lib/src/cmr/umm_spec/location_keywords.clj

```diff
@@ -43,20 +43,15 @@
 (defn- location-values
   "Returns the location keyword values in order so that we can get the last one"
   [location-keyword]
-  (for [k location-keyword-order
-        :let [value (get location-keyword k)]
-        :when value]
-    value))
-
-(defn- leaf-value
-  "Returns the leaf value of the location-keyword object to be put in a SpatialKeywords list"
-  [location-keyword]
-  (last (location-values location-keyword)))
+  (as-> location-keyword keyword
+        (vals keyword)
+        (distinct keyword)
+        (remove nil? keyword)))
 
 (defn location-keywords->spatial-keywords
   "Converts a list of LocationKeyword maps to a list of SpatialKeywords"
   [location-keyword-list]
-  (map #(leaf-value %) location-keyword-list))
+  (flatten (map location-values location-keyword-list)))
```

Comment:
I don't see the three acceptance criteria tested:

1. wildcard searches return values regardless of case (add a case searching for aMEricA or something crazy)
2. first term in Location Keyword hierarchy is returned in a search (add a case searching for Continent that returns all the Continent collections)
3. all records which match search criteria are returned in result set (covered by the test above).